### PR TITLE
Fixes the baseturf bad arg runtime and fixes the jumpToNullSpace proc

### PR DESCRIPTION
**Pull request by ninjanomnom**
image
image

# W H A T

### CODE REVIEW

File: code/modules/shuttle/shuttle.dm [hunk 1]
```diff
@@ -394,28 +394,30 @@
...
+		oldT.change_area(old_area, underlying_area)
+	to_chat(world, num2text(i))
+	for(i in 1 to old_turfs.len)
+		to_chat(world, "TAKE [num2text(i)]")
+		var/turf/oldT = old_turfs[i]
+		if(!oldT)
 			continue
-		T.empty(turf_type, baseturf_type)
+		oldT.empty(turf_type, baseturf_type)
+	to_chat(world, num2text(i))
```
Comment (ninjanomnom) [inline code/modules/shuttle/shuttle.dm]: This proc is somehow failing before it gets to this final to_chat and also as shown in the picture fails halfway through the second loop without a runtime.

File: code/modules/shuttle/shuttle.dm [hunk 2]
```diff
@@ -394,28 +394,30 @@
...
-				underlying_area.contents += T
-				T.change_area(old_area, underlying_area)
-
-	for(var/i in shuttle_turfs)
-		var/turf/T = i
-		if(!T)
+			underlying_area_type = current_dock.area_type
+
+	var/list/old_turfs = return_ordered_turfs(x, y, z, dir, area_type)
+	var/area/underlying_area = locate("[underlying_area_type]")
```
Comment (lzimann) [inline code/modules/shuttle/shuttle.dm]: why are you doing `locate` with a string? Also `locate(type) in GLOB.sortedAreas`

Comment (ninjanomnom) [inline code/modules/shuttle/shuttle.dm]: this is just some code moved over from whoever wrote it originaly

**Comment by ninjanomnom** [comment]
fuck this bug fuck byond for making me manualy error check all the code for something that should be a runtime

**Comment by ninjanomnom** [comment]
also in hindsight it was a bad idea to trust someone to fix a typecache proc who had only used a typecache once incorrectly.

File: code/__HELPERS/_lists.dm [hunk 3]
```diff
@@ -89,13 +89,12 @@
...
-/proc/typecache_filter_list(list/atoms, list/typecache, reversed=FALSE)
+//if reversed, return a new list with only atoms that aren't in typecache L
+/proc/typecache_filter_list(list/atoms, list/typecache, reversed)
 	. = list()
 	for (var/thing in atoms)
 		var/atom/A = thing
-		if (typecache[A.type] && !reversed)
-			. += A
-		else if(reversed)
+		if(typecache[A.type] != reversed) //This assumes typecache[A.type] is either null or TRUE. God help you if it's FALSE
```
Comment (ninjanomnom) [inline code/__HELPERS/_lists.dm]: I make an assumption here that may be an issue if incorrect

**Pull request merged**